### ISSUE TITLE
[chassis] [masic] route flap tests not supporting masic

### ISSUE THREAD

**Issue by wenyiz2021** [sonic-net/sonic-mgmt#7507]
**Description**
https://github.com/sonic-net/sonic-mgmt/pull/5204 added this new testcase that is multi-asic aware

**Steps to reproduce the issue:**
1. run test_route_flap.py on any multi-asic platform
2.
3.

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important:**

    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

**Comment by abdosi** [comment]
@wenyiz2021 is working on this.

**Comment by abdosi** [comment]
test needs enhancement for multi-asic.

**Comment by abdosi** [comment]
Please provide updated PR link is description.

**Comment by wenyiz2021** [comment]
> Please provide updated PR link is description.

updated PR description

**Comment by ysmanman** [comment]
Hi @wenyiz2021 , we patched https://github.com/sonic-net/sonic-mgmt/pull/7557, and test still failed with the following error:  
```
<?xml version="1.0" encoding="utf-8"?><testsuites><testsuite errors="0" failures="1" hostname="71e9d273073f" name="pytest" skipped="0" tests="1" time="2064.958" timestamp="2023-03-20T17:53:28.451020"><properties><property name="topology" value=\
"t2"/><property name="testbed" value="ardut"/><property name="timestamp" value="2023-03-20 17:54:09.656966"/><property name="host" value="cmp227-4"/><property name="asic" value="broadcom"/><property name="platform" value="x86_64-arista_7800r3ak\
_36dm2_lc"/><property name="hwsku" value="Arista-7800R3AK-36DM2-C36"/><property name="os_version" value="branch.202205-ars.12d01cf9-buildimage.origin.202205-nightly-2023.03.18.06.20"/></properties><testcase classname="route.test_route_flap" fil\
e="route/test_route_flap.py" line="174" name="test_route_flap[0]" time="2047.782"><properties><property name="start" value="2023-03-20 17:53:29.673741"/><property name="end" value="2023-03-20 18:27:50.670566"/></properties><failure message="Ass\
ertionError: Did not receive expected packet on any of ports [1, 23, 0, 31, 13, 30, 12, 21, 15, 18, 26, 20, 28, 7, 16, 9, 4, 2, 24, 25, 11, 10, 14, 19, 17, 6, 29, 27, 22, 5, 8, 3] for device 0.                                                    
========== EXPECTED ==========                                                                                                                                                                                                                       
Mask:                                                                                                                                                                                                                                                
0000  94 8E D3 5E 85 04 94 8E D3 5E 85 04 08 00 45 00  ...^.....^....E.                                                                                                                                                                              
0010  00 2E 00 01 00 00 40 01 E3 22 AC 1E 48 24 C0 E8  ......@..&quot;..H$..                                                                                                                                                                         
0020  E2 80 08 00 46 4E 00 00 00 00 30 30 30 30 30 30  ....FN....000000                                                                                                                                                                              
0030  30 30 30 30 30 30 30 30 30 30 30 30              000000000000                                                                                                                                                                                  
mask = 00 00 00 00 00 00 ff ff  ff ff ff ff ff ff ff 00                                                                                                                                                                                              
0010   00 00 00 00 00 00 00 ff  00 00 ff ff ff ff ff ff                                                                                                                                                                                              
0020   ff ff ff 00 00 00 00 00  00 00 ff ff ff ff ff ff                                                                                                                                                                                              
0030   ff ff ff ff ff ff ff ff  ff ff ff ff                                                                                                                                                                                                          
                                                                                                                                                                                                                                                     
========== RECEIVED ==========                                                                                                                                                                                                                       
39561 total packets. Displaying most recent 39561 packets:                                                                                                                                    
 ```

**Comment by wenyiz2021** [comment]
> Hi @wenyiz2021 , we patched #7557, and test still failed with the following error:
> 
> ```
> <?xml version="1.0" encoding="utf-8"?><testsuites><testsuite errors="0" failures="1" hostname="71e9d273073f" name="pytest" skipped="0" tests="1" time="2064.958" timestamp="2023-03-20T17:53:28.451020"><properties><property name="topology" value=\
> "t2"/><property name="testbed" value="ardut"/><property name="timestamp" value="2023-03-20 17:54:09.656966"/><property name="host" value="cmp227-4"/><property name="asic" value="broadcom"/><property name="platform" value="x86_64-arista_7800r3ak\
> _36dm2_lc"/><property name="hwsku" value="Arista-7800R3AK-36DM2-C36"/><property name="os_version" value="branch.202205-ars.12d01cf9-buildimage.origin.202205-nightly-2023.03.18.06.20"/></properties><testcase classname="route.test_route_flap" fil\
> e="route/test_route_flap.py" line="174" name="test_route_flap[0]" time="2047.782"><properties><property name="start" value="2023-03-20 17:53:29.673741"/><property name="end" value="2023-03-20 18:27:50.670566"/></properties><failure message="Ass\
> ertionError: Did not receive expected packet on any of ports [1, 23, 0, 31, 13, 30, 12, 21, 15, 18, 26, 20, 28, 7, 16, 9, 4, 2, 24, 25, 11, 10, 14, 19, 17, 6, 29, 27, 22, 5, 8, 3] for device 0.                                                    
> ========== EXPECTED ==========                                                                                                                                                                                                                       
> Mask:                                                                                                                                                                                                                                                
> 0000  94 8E D3 5E 85 04 94 8E D3 5E 85 04 08 00 45 00  ...^.....^....E.                                                                                                                                                                              
> 0010  00 2E 00 01 00 00 40 01 E3 22 AC 1E 48 24 C0 E8  ......@..&quot;..H$..                                                                                                                                                                         
> 0020  E2 80 08 00 46 4E 00 00 00 00 30 30 30 30 30 30  ....FN....000000                                                                                                                                                                              
> 0030  30 30 30 30 30 30 30 30 30 30 30 30              000000000000                                                                                                                                                                                  
> mask = 00 00 00 00 00 00 ff ff  ff ff ff ff ff ff ff 00                                                                                                                                                                                              
> 0010   00 00 00 00 00 00 00 ff  00 00 ff ff ff ff ff ff                                                                                                                                                                                              
> 0020   ff ff ff 00 00 00 00 00  00 00 ff ff ff ff ff ff                                                                                                                                                                                              
> 0030   ff ff ff ff ff ff ff ff  ff ff ff ff                                                                                                                                                                                                          
>                                                                                                                                                                                                                                                      
> ========== RECEIVED ==========                                                                                                                                                                                                                       
> 39561 total packets. Displaying most recent 39561 packets:                                                                                                                                    
> ```

thanks @ysmanman , I starting to see this as well, last month when I create PR 7557 it was fine on the same testbed, I need to check more
